tableauserverclient/server/request_factory.py

```diff
@@ -42,6 +42,7 @@ def _generate_xml(self, datasource_item, connection_credentials=None):
             credentials_element.attrib['name'] = connection_credentials.name
             credentials_element.attrib['password'] = connection_credentials.password
             credentials_element.attrib['embed'] = 'true' if connection_credentials.embed else 'false'
+            credentials_element.attrib['oAuth'] = 'true' if connection_credentials.oauth else 'false'
```

Comment:
Even with my above comment, this would be including the oauth parameter in all requests, even if it's not an oauth enabled connection... Can you test and see if that causes a problem? If it does I suggest defaulting to `None` and then doing something like:
```python
if connection_credentials.oauth is not None:
    redentials_element.attrib['oAuth'] = 'true'
```